### PR TITLE
Lokinet LNS

### PR DESCRIPTION
**Pull request by jagerman**
Lokinet LNS support, activating at HF16.

See individual commit messages for details; the big features:

- Enable .loki registrations at HF16, and update a bunch of things related to how we handle them.
- Replace LNS argon2 encryption with a much lighter version beginning at HF16
- Overhaul (and greatly simplify) how we track and handle LNS updates.
- Lower LNS fees a bit (25%) at HF16 to reflect Loki's recent price change.

### CODE REVIEW

**Comment by jagerman** [comment]
This doesn't yet prune old entries from the database; that can come in another PR (and isn't necessary at all for this to work--it will just reduce how big the lns.db can get).

File: src/cryptonote_core/loki_name_system.cpp [hunk 1]
```diff
@@ -1483,60 +1489,92 @@ CREATE TABLE IF NOT EXISTS "settings" (
...
   {
     MERROR("Can not generate SQL table for LNS: " << (table_err_msg ? table_err_msg : "??"));
     sqlite3_free(table_err_msg);
     return false;
   }
 
+  // In Loki 8 we dropped some columns that are no longer needed, but SQLite can't do this easily:
+  // instead we have to manually recreate the table, so check it and see if the prev_txid or
+  // register_height columns still exist: if so, we need to recreate.
+  bool need_mappings_migration = false;
```
Comment (Doy-lee) [inline src/cryptonote_core/loki_name_system.cpp:1518]: There are 2 places we do migration and its sort of weird that we want to build default tables and then check migration again afterwards.

Feels like what we want is to load the tables, check version then update, avoiding building a default table unless we know we have to (lns.db itself doesn't exist yet).

**Review by Doy-lee — COMMENTED** [review]
Fairly straight foward changes :+1:

File: src/cryptonote_core/loki_name_system.cpp (hunk 1 above)
Comment (jagerman) [reply src/cryptonote_core/loki_name_system.cpp:1518]: > There are 2 places we do migration and its sort of weird that we want to build default tables and then check migration again afterwards.

The reason for doing it this way is that we're doing a `CREATE TABLE IF NOT EXISTS` which means the query will go ahead just fine it already exist with the old structure, but if it *doesn't* then we create it.

The `PRAGMA table_info` should be an extremely fast query, and only done once during initialization to check whether the table we have is the old one (with prev_txid/register_height) or not.

I did not like this messiness: `ALTER TABLE "mappings" DROP COLUMN "prev_txid"` is how I tried to do it at first, only to discover that sqlite3 does not support dropping columns from a table at all (the create-a-new-table-and-copy-it is the documented solution to do this).

Comment (jagerman) [reply src/cryptonote_core/loki_name_system.cpp:1518]: Oh, re: the settings version: I see that as being for data value migration (e.g. where we needed to go compute the update heights by scanning the database), whereas this chunk of code is only for database schema changes.  It just happens that sqlite3 requires a mess to drop a column. :(

Comment (Doy-lee) [reply src/cryptonote_core/loki_name_system.cpp:1518]: Yeah, in the long run it doesn't matter too much, essentially can drop the upgrade code over time as everyone upgrades.

**Pull request merged**